fix: Support GPT-5.1 reasoning.effort='none' in proxy

Chesars opened the pull request:
fix: Support GPT-5.1 reasoning.effort='none' in proxy
  
## Relevant issues

  Fixes #16741 
  Closes #16839

  ## Pre-Submission checklist

  - [ ] I have Added testing in the `tests/litellm/` directory
  - [ ] I have added a screenshot of my new test passing locally
  - [ ] My PR passes all unit tests on `make test-unit`
  - [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

  ## Type

  🐛 Bug Fix

  ## Changes

## Summary

  Updates LiteLLM to use OpenAI SDK 2.8+ which includes native support for GPT-5.1's `reasoning.effort='none'` parameter. This fixes Pydantic validation errors in the **LiteLLM Proxy** when handling GPT-5.1 streaming responses.

  ## Context

  **The Issue (Proxy):**
  When using GPT-5.1 through the LiteLLM Proxy, streaming responses failed with:
  Input should be 'minimal', 'low', 'medium' or 'high' [type=literal_error, input_value='none']

  **Why in the Proxy?**
  - The LiteLLM **Proxy** re-validates response chunks with Pydantic during transformation (`litellm/llms/openai/responses/transformation.py`)
  - The **SDK/Library** doesn't do this validation, so direct SDK usage worked fine
  - GPT-5.1 returns `reasoning.effort='none'`, which wasn't in the Pydantic schema

  **Previous Workaround (PR #16745):**
  Added a custom `Reasoning` TypedDict that included `'none'` to work around OpenAI SDK < 2.8.0 limitations while maintaining compatibility with `semantic-router < 2.0.0`.

 **Note:** semantic-router released v0.1.12 which supports OpenAI SDK 2.x

  ## Changes

  **Dependencies:**
  - ⬆️ Python: `>=3.8.1` → `>=3.9,<4.0` (required by OpenAI SDK 2.8.0+; Python 3.8 EOL Oct 2024)
  - ⬆️ OpenAI SDK: `>=1.99.5` → `>=2.8.0` (includes native `reasoning.effort='none'` support)
  - ⬆️ semantic-router: `*` → `>=0.1.12` with `python >=3.9,<3.14` constraint
  - ⬆️ resend: `^0.8.0` → `>=0.8.0` (allows 2.x versions)

  **Code:**
  - ✅ Removed custom `Reasoning` TypedDict from `litellm/types/llms/openai.py`
  - ✅ Now imports `Reasoning` directly from `openai.types.responses.response_create_params`
  - ✅ Proxy validation now uses OpenAI SDK's native types with `'none'` support

  ## Breaking Changes

  **⚠️ Minimum Python version increased to 3.9**

  This is **required** by OpenAI SDK 2.8.0+. Python 3.8 reached end-of-life in October 2024.

  **Migration:** Users on Python 3.8 can continue using LiteLLM v1.80.x or earlier.

  ## Testing

  - ✅ Verified `Reasoning` type imports successfully from OpenAI SDK 2.8.1
  - ✅ Confirmed type accepts `effort='none'` value for Pydantic validation
  - ✅ resend email tests pass with updated dependencies
  - ✅ No dependency conflicts with `poetry update`

  ## Related

  Fixes #16741

Chesars committed:
fix: Add support for GPT-5.1 reasoning.effort='none' parameter

- Override Reasoning type to include 'none' effort value
- Maintains compatibility with semantic-router (requires openai <2.0.0)
- GPT-5.1 defaults to reasoning.effort='none' as of OpenAI SDK 2.8.0
- Override can be removed once semantic-router supports OpenAI SDK 2.x

Fixes #16741

krrishdholakia commented on litellm/types/llms/openai.py:
```diff
@@ -51,12 +53,38 @@
     ToolParam,
 )
+# Override Reasoning type to support GPT-5.1's "none" effort value
+# OpenAI SDK >=2.8.0 includes this, but we maintain compatibility with <2.0.0 for semantic-router
```
@ishaan-jaff it seems wrong for us to be blocked on openai compatibility b/c of a dependency. Can we move off semantic-router or work around this limitation? 

@Chesars open to your thoughts on this

Chesars committed:
feat: Update to OpenAI SDK 2.8+ and Python 3.9+

- Update minimum Python version from 3.8.1 to 3.9 (3.8 EOL Oct 2024)
- Update OpenAI SDK requirement from >=1.99.5 to >=2.8.0
- Update semantic-router to >=0.1.12 with Python >=3.9,<3.14 constraint
- Update resend to >=0.8.0 (allows 2.x versions)
- Revert custom Reasoning TypedDict - now using native type from OpenAI SDK 2.8+

OpenAI SDK 2.8.0 natively supports reasoning.effort='none' for GPT-5.1,
eliminating the need for our custom type override.

Chesars committed:
chore: Update poetry.lock with OpenAI SDK 2.8.1 and dependencies
